civihr_employee_portal/updates/7044.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/**
+ * Updates contact address, enforcing only one personal address type
+ */
+function civihr_employee_portal_update_7044() {
+  civicrm_initialize();
+  $locationType = civicrm_api3('LocationType', 'get', ['name' => 'Work']);
+  if (!$locationType['count']) {
+    return;
+  }
+
+  $addresses = _get_personal_address_count();
+  foreach ($addresses as $address) {
+    if ($address['total'] < 2) {
+      continue;
+    }
+
+    $result = civicrm_api3('Address', 'get', [
+      'location_type_id' => 'Personal',
+      'contact_id' => $address['contact_id']
+    ]);
+
+    $contactAddresses = array_values($result['values']);
+    $primaryIndex = array_search('1', array_column($contactAddresses, 'is_primary'));
+    if ($primaryIndex === FALSE) {
+      $isWorkPrimary = _check_contact_work_address_is_primary($address['contact_id']);
```

Comment:
If contact currently have a Work location type set as Primary, then no need to update the first item.